lib/kafka/cluster.rb

```diff
@@ -46,6 +46,10 @@ def add_target_topics(topics)
         @target_topics.merge(new_topics)
 
         refresh_metadata!
+
+        cluster_info.topics.each do |topic|
+          Protocol.handle_error(topic.topic_error_code)
+        end
```

Comment:
Indeed, it does break in the specs. LeaderNotAvailable is raised until the leader is, well, available. But the user code should not break because of that.